lnd.go

```diff
@@ -246,50 +246,21 @@ func Main(cfg *Config, lisCfg ListenerCfg, shutdownChan <-chan struct{}) error {
 		defer pprof.StopCPUProfile()
 	}
 
-	// Create the network-segmented directory for the channel database.
-	ltndLog.Infof("Opening the main database, this might take a few " +
-		"minutes...")
-
-	startOpenTime := time.Now()
-
 	ctx := context.Background()
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	if cfg.DB.Backend == lncfg.BoltBackend {
-		ltndLog.Infof("Opening bbolt database, sync_freelist=%v",
-			cfg.DB.Bolt.SyncFreelist)
-	}
-
-	databaseBackends, err := cfg.DB.GetBackends(
-		ctx, cfg.localDatabaseDir(), cfg.networkName(),
-	)
-	if err != nil {
-		ltndLog.Error(err)
-		return err
-	}
-
-	// Open the channeldb, which is dedicated to storing channel, and
-	// network related metadata.
-	chanDB, err := channeldb.CreateWithBackend(
-		databaseBackends.LocalDB,
-		channeldb.OptionSetRejectCacheSize(cfg.Caches.RejectCacheSize),
-		channeldb.OptionSetChannelCacheSize(cfg.Caches.ChannelCacheSize),
-		channeldb.OptionDryRunMigration(cfg.DryRunMigration),
-	)
+	localChanDB, remoteChanDB, err := initializeDatabases(ctx, cfg)
 	switch {
 	case err == channeldb.ErrDryRunMigrationOK:
 		ltndLog.Infof("%v, exiting", err)
 		return nil
-
 	case err != nil:
-		ltndLog.Errorf("Unable to open channeldb: %v", err)
-		return err
+		return fmt.Errorf("unable to open databases: %v", err)
 	}
-	defer chanDB.Close()
 
-	openTime := time.Since(startOpenTime)
-	ltndLog.Infof("Database now open (time_to_open=%v)!", openTime)
+	defer localChanDB.Close()
+	defer remoteChanDB.Close()
```

Comment:
is it safe to call `Close()` twice on the local db?

Reply:
Didn't see any issues w/ it locally, but modified things to pass a composite "clean up" function instead, which hides more of the details from the caller.

Reply:
is the latest version pushed up? still looks the same to me

Reply:
Oops, just pushed it up, PTAL!